src/config.mts

```diff
@@ -0,0 +1,41 @@
+import path from 'path';
+import { pathToFileURL } from 'url';
+
+let configurationInstance: Config | undefined = undefined;
+
+interface IOConfig {
```

Comment:
Types like this one should be defined in other modules with specific logic that utilize the types. In other words, "configuration" is just an aggregation of initial state for the tool.